baremaps-openapi/src/main/java/com/baremaps/openapi/services/TilesetsService.java

```diff
@@ -0,0 +1,71 @@
+package com.baremaps.openapi.services;
+
+import com.baremaps.api.TilesetsApi;
+import com.baremaps.model.TileSet;
+import java.util.List;
+import java.util.UUID;
+import org.jdbi.v3.core.Jdbi;
+import org.jdbi.v3.core.qualifier.QualifiedType;
+import org.jdbi.v3.jackson2.Jackson2Plugin;
+import org.jdbi.v3.json.Json;
+import org.jdbi.v3.postgres.PostgresPlugin;
+
+public class TilesetsService implements TilesetsApi {
+
+  private Jdbi jdbi;
+
+  public TilesetsService() {
+    this.jdbi = Jdbi.create("jdbc:postgresql://localhost:5432/baremaps?user=baremaps&password=baremaps");
+    this.jdbi.installPlugin(new PostgresPlugin());
+    this.jdbi.installPlugin(new Jackson2Plugin());
+  }
+
+  QualifiedType<TileSet> qualifiedType = QualifiedType.of(TileSet.class).with(Json.class);
+
+  @Override
+  public void addTileset(TileSet tileset) {
+    UUID tilesetId = UUID.randomUUID(); // TODO: Read from body
+    jdbi.useHandle(handle -> {
+      handle.createUpdate("insert into tilesets (id, tileset) values (:id, :json)")
+          .bindByType("json", tileset, qualifiedType)
+          .bind("id", tilesetId.toString())
+          .execute();
+    });
+  }
+
+  @Override
+  public void deleteTileset(String tilesetId) {
+    jdbi.useHandle(handle -> {
+      handle.execute("delete from tilesets where id = (?)", tilesetId);
+    });
+  }
+
+  @Override
+  public TileSet getTileset(String tilesetId) {
```

Comment:
I tested the service with a column of type `jsonb` and it seems to be working:

```
create table tilesets
(
    id      text,
    tileset jsonb
);
```
Is i the way you intent to use it?

Reply:
Yes this is what we go for at the moment. It may change in the future.